app/src/test/java/waiter/CommunicatorTest.java

```diff
@@ -0,0 +1,21 @@
+package waiter;
+
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class CommunicatorTest {
+
+    @Test
+    void establishesCommunication() throws IOException {
+        String userInput = "Doesn't matter whats here, simulating any user input message";
```

Comment:
```suggestion
        String userInput = "foo";
```

Non-blocking: I would typically just use something like "foo" here. Most developers won't assume that there is some special behavior when "foo", "bar", "baz", etc strings are encountered in tests.

Reply:
Another option would be to parameterize this test to indicate in the test that it doesn't matter what user input is entered ("foo", "bar", "baz"), we just care that it is equal to `calledWithConnection.read().open()`. You could take this a step further with "property-based testing" or "fuzz testing".

Reply:
I shall look into this! Thank you 😄

Reply:
Interested in this too - could you post your findings here Jon?

Reply:
Hmmm, so from what I understand and looked up, generally this is done by either manually creating a list of inputs or using a randomly generated list of inputs (via a library or something), picking one of those inputs randomly and then testing using that input since we are just checking for equivalence. Also I do would do this random input generation and selection in the test, right?

Reply:
I see a couple of options here:

1. Change the string name to something like "foo" to indicate that this is some arbitrary string.
2. Use a library like [Faker](https://github.com/DiUS/java-faker) to generate a random fake string, just be careful that you aren't introducing an intermittently failing test (sometimes referred to as a flaky test).
3. Write a parameterized test to prove that this works for a relatively small, distinct set of strings (not what I would reach for in this particular case).
4. Write a property-based test to prove that this works for a set of inputs that abide by some preconditions (i.e. all `Strings` limited to ASCII characters, with a length of 0-20, that don't contain "\n").

Reply:
I have changed it to "foo" for now, I want to try and implement a property-based test as an improvement.

Reply:
Resolved in [this commit](https://github.com/jonmanahan/waiter/pull/2/commits/e53fdbc1eec22480eae57e79838d0f1186c16f90) via changing the `userinput` to "foo", but also want to continue this issue into a future PR as well to implement property-based testing. Potential ticket name -> property-based-testing ticket, thank you all for the help!